### ISSUE TITLE
[Accessibility Visual Requirement] - No tooltips for starred dashboards in Dashlist component (MAS ID: 260681)

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#72834]
**Steps to reproduce**:
Open #Url: https://play.grafana.org/. Sign in with valid credentials. "Home" page appears. Navigate the Feature Showcase pane heading and select it. Navigate the expanded Dropdown and select the view menu item. Navigate the page "General" page which appears. Verify the issue.

**Actual Result**:
Tooltip is not defined for 'Star dashboard 1 time series graph' checkbox under 'General' screen.

**Expected Result**
 Tooltip should be defined for 'Star dashboard 1 time series graph' checkbox under 'General' screen. Tooltip should be defined as 'Star dashboard 1'. 
 
**Relevant WCAG Criteria:**
Could possibly make an argument for [3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.1#labels-or-instructions)?

![image](https://github.com/grafana/grafana/assets/6906380/a61a060c-755a-43dd-bc81-56440c6c88b2)

**Component**
`<DashList />`
